### ISSUE TITLE
error when DataLoader's num_workers is not 0

### ISSUE THREAD

**Issue by mengjiexu** [apache/incubator-mxnet#13313]
when I use:
train_data = gluon.data.DataLoader(
        MyImageFolderDataset(label_len).transform_first(transform_train),
        batch_size=batch_size,num_workers=4)
It shows error:
      Can't pickle local object 'Dataset.transform_first.<locals>.base_fn'

when I use:
train_data = gluon.data.DataLoader(
        MyImageFolderDataset(label_len).transform_first(transform_train),
        batch_size=batch_size,**num_workers=0**)
It is ok.

**Comment by zachgk** [comment]
@mxnet-label-bot add [Bug, Python, Gluon, Data-loading]

Thank you for submitting the issue! I'm labeling it so the MXNet community members can help resolve it.

**Comment by intgogo** [comment]
Same problem in windows server 2016.

**Comment by gaomingnudt** [comment]
I got the same problem on Win10
mxnet is latest version
and the same code work fine on mac

**Comment by abhinavs95** [comment]
@mengjiexu looks like #14541  is a duplicate issue of this. Acc @zhreshold comment there the issue is fixed in 1.4.0 and master
